pkg/virt-handler/cgroup/util.go

```diff
@@ -0,0 +1,89 @@
+package cgroup
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"syscall"
+
+	runc_configs "github.com/opencontainers/runc/libcontainer/configs"
+
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	cgroupStr = "cgroup"
+
+	procMountPoint = "/proc"
+
+	HostRootPath       = "/proc/1/root"
+	cgroupBasePath     = "/sys/fs/" + cgroupStr
+	HostCgroupBasePath = HostRootPath + cgroupBasePath
+)
+
+// Templates for logging / error messages
+const (
+	errApplyingDeviceRule = "error occurred while applying device rule: %v"
+	errApplyingOtherRules = "error occurred while applying rules (that are not device rules): %v"
+	settingDeviceRule     = "setting device rule for cgroup %s: %v"
+
+	v1Str = "v1"
+	v2Str = "v2"
+
+	loggingVerbosity = 2
+)
+
+// getNewResourcesWithoutDevices returns a new Resources struct with Devices attributes dropped
+func getNewResourcesWithoutDevices(r *runc_configs.Resources) runc_configs.Resources {
+	resourcesWithoutDevices := *r
+	resourcesWithoutDevices.Devices = nil
+
+	return resourcesWithoutDevices
+}
+
+func logAndReturnErrorWithSprintfIfNotNil(err error, template string, args ...interface{}) error {
+	if err == nil {
+		return nil
+	}
+
+	newErr := fmt.Errorf(template, args...)
+	log.Log.Error(newErr.Error())
+	return newErr
+}
+
+func areResourcesEmpty(r runc_configs.Resources) bool {
+	return reflect.DeepEqual(r, runc_configs.Resources{})
+}
+
+// RunWithChroot changes the root directory (via "chroot") into newPath, then
+// runs toRun function. When the function finishes, changes back the root directory
+// to the original one that
+func RunWithChroot(newPath string, toRun func() error) error { // ihol3 bad place to define func
```

Comment:
Yes, but it's there for a good reason :)
Can you look at `Special notes for your reviewer:` number (2)?
I've tried to place this in a more global place (since this is not related only to cgroups) but had problems with that.. Maybe you can help